### ISSUE TITLE
JAIF-based whole program inference does not correctly handle conflicting annotation names

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#3009]
Discovered while testing stub-backed WPI (#2871).

The failing test is `ConflictingAnnotationsTest.java`, which will be merged with #2871 but disabled for jaif-based WPI:

```
// Tests whether inferring an @Sibling1 annotation when another
// @Sibling1 annotation from a different namespace is present causes
// problems.

class ConflictingAnnotationsTest {

    int getWPINamespaceSibling1() {
        return MethodReturnTest.getSibling1();
    }

    // This version of Sibling1 is not typechecked - it doesn't
    // belong to the checker and instead is defined in the Sibling1.java
    // file in this directory.
    @Sibling1 int getLocalSibling1() {
        return 1;
    }
}
```

There is also the definition of the alternative `@Sibling1` annotation in the WPI test directory.

JAIF-based WPI blindly imports `testlib.wholeprograminference.qual.Sibling1` annotation, causing a conflicting annotations error to be issued on the return type of the `getLocalSibling1` method (which also has `WholeProgramInferenceBottom` inferred.

The workaround to prevent the jaif-based version from running on this test is to delete the test from the `annotated` subdirectory before checking for errors after WPI runs.

**Comment by kelloggm** [comment]
Here is another test that JAIF-based WPI fails discovered while testing stub-based WPI (`MultiDimensionalArrays.java`):

```java
// This test ensures that annotations on different component types of multidimensional arrays
// are printed correctly.

import testlib.wholeprograminference.qual.Sibling1;
import testlib.wholeprograminference.qual.Sibling2;

class MultiDimensionalArrays {

    // two dimensional arrays

    void requiresS1S2(@Sibling1 int @Sibling2 [] x) {}

    int[] twoDimArray;

    void testField() {
        // :: error: argument.type.incompatible
        requiresS1S2(twoDimArray);
    }

    void useField(@Sibling1 int @Sibling2 [] x) {
        twoDimArray = x;
    }

    void testParam(int[] x) {
        // :: error: argument.type.incompatible
        requiresS1S2(x);
    }

    void useParam(@Sibling1 int @Sibling2 [] x) {
        testParam(x);
    }

    // three dimensional arrays

    void requiresS1S2S1(@Sibling1 int @Sibling2 [] @Sibling1 [] x) {}

    int[][] threeDimArray;

    void testField2() {
        // :: error: argument.type.incompatible
        requiresS1S2S1(threeDimArray);
    }

    void useField2(@Sibling1 int @Sibling2 [] @Sibling1 [] x) {
        threeDimArray = x;
    }

    void testParam2(int[][] x) {
        // :: error: argument.type.incompatible
        requiresS1S2S1(x);
    }

    void useParam2(@Sibling1 int @Sibling2 [] @Sibling1 [] x) {
        testParam2(x);
    }

    // three dimensional array with annotations only on two inner types

    void requiresS1S2N(@Sibling1 int @Sibling2 [][] x) {}

    int[][] threeDimArray2;

    void testField3() {
        // :: error: argument.type.incompatible
        requiresS1S2N(threeDimArray2);
    }

    void useField3(@Sibling1 int @Sibling2 [][] x) {
        threeDimArray2 = x;
    }

    void testParam3(int[][] x) {
        // :: error: argument.type.incompatible
        requiresS1S2N(x);
    }

    void useParam3(@Sibling1 int @Sibling2 [][] x) {
        testParam3(x);
    }

    // three dimensional array with annotations only on two array types, not innermost type

    void requiresS2S1(int @Sibling2 [] @Sibling1 [] x) {}

    int[][] threeDimArray3;

    void testField4() {
        // :: error: argument.type.incompatible
        requiresS2S1(threeDimArray3);
    }

    void useField4(int @Sibling2 [] @Sibling1 [] x) {
        threeDimArray3 = x;
    }

    void testParam4(int[][] x) {
        // :: error: argument.type.incompatible
        requiresS2S1(x);
    }

    void useParam4(int @Sibling2 [] @Sibling1 [] x) {
        testParam4(x);
    }
}
```

JAIF WPI fails to insert the correct annotations for 3+ dimensional arrays.
